connectors/src/lib/redis.ts

```diff
@@ -1,16 +1,70 @@
+import type { RedisClientType } from "redis";
 import { createClient } from "redis";
 
-export async function redisClient() {
-  const { REDIS_URI } = process.env;
-  if (!REDIS_URI) {
-    throw new Error("REDIS_URI is not defined");
-  }
-  const client = createClient({
-    url: REDIS_URI,
-  });
-  client.on("error", (err) => console.log("Redis Client Error", err));
+import logger from "@connectors/logger/logger";
+
+let client: RedisClientType;
+
+export async function redisClient(): Promise<RedisClientType> {
+  if (!client) {
+    const { REDIS_URI } = process.env;
+    if (!REDIS_URI) {
+      throw new Error("REDIS_URI is not defined");
+    }
 
-  await client.connect();
+    client = createClient({
+      url: REDIS_URI,
```

Comment:
Missed this in the first review, but introducing a redis pool here requires more work, as we need to ensure that we remove all the call to `.quit()` in the existing code.